# config/vlp16_lexus.param.yaml
---

lidars:
  ray_ground_classifier:
    ros__parameters:
      timeout_ms:       10
      cloud_timeout_ms: 110
      raw_topic:       "points_filtered"
      ground_topic:    "points_ground"
      nonground_topic: "nonground_points"
      pcl_size:         55000
      frame_id:        "base_link"
      is_structured:    true
      classifier:
        sensor_height_m:                     0.368
        max_local_slope_deg:                 20.0
        max_global_slope_deg:                7.0
        nonground_retro_thresh_deg:          70.0
        min_height_thresh_m:                 0.05
        max_global_height_thresh_m:          0.3
        max_last_local_ground_thresh_m:      0.6
        max_provisional_ground_distance_m:   5.0
        min_height_m:                        -0.5
        max_height_m:                        1.5
      aggregator:
        min_ray_angle_rad: -3.14159
        max_ray_angle_rad:  3.14159
        ray_width_rad:      0.01
        max_ray_points:     512
